electron/src/main.ts

```diff
@@ -283,6 +304,48 @@ const handleAppEvents = () => {
     isQuitting = true;
   });
 
+  app.on('login', async (event, webContents, request, authInfo, callback) => {
+    if (authInfo.isProxy) {
+      event.preventDefault();
+
+      if (authenticatedProxyInfo && authenticatedProxyInfo.username && authenticatedProxyInfo.password) {
+        logger.info('Sending provided credentials to authenticated proxy ...');
+        return callback(authenticatedProxyInfo.username, authenticatedProxyInfo.password);
+      }
+
+      const systemProxy = await getProxySettings();
+      const systemProxySettings = systemProxy && (systemProxy.http || systemProxy.https);
+
+      if (systemProxySettings) {
+        authenticatedProxyInfo = new URL(
+          `http://${systemProxySettings.credentials.username}:${systemProxySettings.credentials.password}@${authInfo.host}:${authInfo.port}`,
```

Comment:
We should respect the system-wide settings regarding the protocol. http is used here but it could be a https proxy. Also what about socks5://... etc?